src/fides/api/api/v1/endpoints/privacy_notice_endpoints.py

```diff
@@ -227,140 +214,14 @@ def create_privacy_notices(
     To avoid any confusing or unexpected behavior, the entire operation is void
     if any of the input data does not satisfy validation criteria.
     """
-    validate_notice_data_uses(privacy_notices, db)
-
-    existing_notices = (
-        PrivacyNotice.query(db).filter(PrivacyNotice.disabled.is_(False)).all()
-    )
-
-    new_notices = [
-        PrivacyNotice(**privacy_notice.dict(exclude_unset=True))
-        for privacy_notice in privacy_notices
-    ]
     try:
-        check_conflicting_data_uses(new_notices, existing_notices)
-    except ValidationError as e:
-        raise HTTPException(HTTP_422_UNPROCESSABLE_ENTITY, detail=e.message)
-
-    # Loop through and create the new privacy notices
-    created_privacy_notices: List[PrivacyNotice] = []
-    affected_regions: Set = set()
-    for privacy_notice in privacy_notices:
-        privacy_notice = transform_fields(
-            transformation=escape,
-            model=privacy_notice,
-            fields=ESCAPE_FIELDS,
-        )
-        created_privacy_notice = PrivacyNotice.create(
-            db=db,
-            data=privacy_notice.dict(exclude_unset=True),
-            check_name=False,
-        )
-        created_privacy_notices.append(created_privacy_notice)
-        affected_regions.update(created_privacy_notice.regions)
-
-    # After created all notices, make sure experiences exist to back all notices.
-    upsert_privacy_experiences_after_notice_update(
-        db, affected_regions=list(affected_regions)
-    )
+        created_privacy_notices, _ = create_privacy_notices_util(db, privacy_notices)
+    except (ValueError, ValidationError) as e:
+        raise HTTPException(HTTP_422_UNPROCESSABLE_ENTITY, detail=str(e))
 
     return created_privacy_notices
 
 
-def validate_privacy_notice_dry_update(dry_update: PrivacyNotice) -> None:
-    """
-    Verify that a dry update of a PrivacyNotice satisfies the constraints
-    for creating a privacy notice.
-
-    This is done here instead of upfront because we need access to the current values
-    of the privacy notice in the database combined with the patch updates from the request
-    """
-    try:
-        PrivacyNoticeCreation.from_orm(dry_update)
-    except ValueError as exc:
-        raise HTTPException(
-            status_code=HTTP_422_UNPROCESSABLE_ENTITY,
-            # pylint: disable=no-member
-            detail=exc.errors(),  # type: ignore
-        )
-
-
-def prepare_privacy_notice_patches(
-    privacy_notice_updates: List[schemas.PrivacyNoticeWithId],
-    db: Session,
-) -> List[Tuple[schemas.PrivacyNoticeWithId, PrivacyNotice]]:
```

Comment:
Moved to src/fides/api/util/consent_util.py and heavily adapted in several places to share between updating privacy notices and new updating privacy templates on startup.